crates/file_store/src/store.rs

```diff
@@ -109,6 +117,19 @@ where
         }
     }
 
+    /// Iterates over the stored **versioned** changesets from first to last, changing the seek
+    /// position at each iteration.
+    ///
+    /// The iterator may fail to read an entry and therefore return an error. However, the first time
+    /// it returns an error will be the last. After doing so, the iterator will always yield `None`.
+    ///
+    /// **WARNING**: This method changes the write position in the underlying file. You should
+    /// always iterate over all entries until `None` is returned if you want your next write to go
+    /// at the end; otherwise, you will write over existing entries.
+    pub fn iter_versioned_changesets(&mut self) -> EntryIter<V> {
+        EntryIter::new(self.magic_len as u64, &mut self.db_file)
+    }
+
```

Comment:
`iter_changesets` should be built from `iter_versioned_changesets`. We only want to read versioned changesets directly from the file.

Reply:
```rust
    pub fn iter_changesets(&mut self) -> impl Iterator<Item = Result<C, IterError>> + '_ {
        self.iter_versioned_changesets().map(|r| r.map(Into::into))
    }
```